test/test_dataloader.py

```diff
@@ -434,6 +434,38 @@ def _test_batch_sampler(self, **kwargs):
                 self.assertEqual(len(input), 3)
                 self.assertEqual(input, self.data[offset:offset + 3])
 
+    def test_RandomSampler(self):
+
+        def sample_stat(sampler, num_samples):
+            bin = {}
+            minval, maxval = num_samples, 0
+            for s in sampler:
+                bin[s] = bin.get(s, 0) + 1
+                minval = min(minval, s)
+                maxval = max(maxval, s)
+
+            count_repeated = 0
+            for val in bin.values():
+                if val > 1:
+                    count_repeated += 1
+            return (count_repeated, minval, maxval)
+
+        from torch.utils.data import RandomSampler
+        n = len(self.dataset)
+        sampler_with_replacement = RandomSampler(self.dataset, n, True)
+        count_repeated, minval, maxval = sample_stat(sampler_with_replacement, n)
+        self.assertTrue(count_repeated > 0)
```

Comment:
In here `n` should be at least `len(self.dataset) + 1`. Otherwise you can end up with every index showing up exactly once, causing this condition to fail, and this test to be flaky.

Reply:
Nice catch! I will change it